src/styles/shared/components/buttons/_config.scss

```diff
@@ -0,0 +1,75 @@
+@use 'sass:map';
+@use '../../variables' as var;
+
+// Map variables to local tokens
+$touch-target-size: var.$touch-target-size !default;
+
+$color-g1: var.$color-grey-1 !default;
+$color-g2: var.$color-grey-2 !default;
+$color-g3: var.$color-grey-3 !default;
+$color-g6: var.$color-grey-6 !default;
+$color-g7: var.$color-grey-7 !default;
+$color-gsemi: var.$color-grey-semi !default;
+$color-gmid: var.$color-grey-mid !default;
+$color-brand: var.$color-brand !default;
+$color-link-hover: var.$color-link-hover !default;
+
+// Define "colormaps" to style buttons with
+
+// Default colors for buttons
+$colors: (
+  'foreground': $color-gmid,
+  'background': $color-g1,
+  'hover-foreground': $color-g7,
+  'hover-background': $color-g2,
+  'active-foreground': $color-g7,
+  'active-background': $color-g1,
+  'border': transparent,
+  'disabled-foreground': $color-gmid,
+);
+
+// Icon-only buttons have a transparent background by default, and a more
+// visible active (pressed) state
+$IconButton-colors: map.merge(
+  $colors,
+  (
+    'background': transparent,
+    'hover-background': transparent,
+    'active-foreground': $color-brand,
+    'active-background': transparent,
+  )
+);
+
+$IconButton-colors--primary: map.merge(
+  $IconButton-colors,
+  (
+    'foreground': $color-brand,
+    'hover-foreground': $color-brand,
+    'active-foreground': $color-brand,
+  )
+);
+
+$LabeledButton-colors: $colors;
+
+// This set of colors is light text on dark background
+$LabeledButton-colors--primary: map.merge(
+  $colors,
```

Comment:
I established a convention of only extended the base color maps (not extending maps that extend those) because of that multiple-indirection you mention. I want it to be relatively easy to see how each color map differs from the base colors, without having to factor extra levels in. I toyed with the idea of making every color map here _complete_ (versus using `map.merge` to extend the base colors), but stopped short of that as a compromise.